scripts/README.md

```diff
@@ -0,0 +1,255 @@
+# scripts
+
+The TANF app uses several scripts through its lifecycle. 
+These don't all get used by or interacted with by us too often,
+but some are mission critical during deployment and review of 
+the application. When a developer is working on these scripts, 
+they should update this documentation so future developer's 
+understand the role of the scripts
+
+# set-backend-env-vars.sh 
+
+## usage 
+
+```bash
+
+./scripts/set-backend-env-vars.sh CGAPPNAME_BACKEND CG_SPACE
+```
+
+## args
+
+`CGAPPNAME_BACKEND` the sub domain of the app you are trying to set up environment variables for.
+`CG_SPACE` the space the app you are trying to set up for exists in.
+
+## description
+
+Determine the appropriate BASE_URL for the deployed instance based on the
+provided Cloud.gov App Name
+Use Shell Parameter Expansion to replace localhost in the URL
+Dynamically generate a new `DJANGO_SECRET_KEY`
+Dynamically set `DJANGO_CONFIGURATION` based on Cloud.gov Space
+
+# sudo-check.sh
+
+## usage
+
+```
+./scripts/sudo-check.sh
+
+```
+
+## Args
+
+no args
+
+## description
+
+check if the `sudo` command is present, installs it if it isn't.
+
+# cf-checks.sh
+
+## usage
+
+```bash
+./scripts/cf-check.sh
+```
+
+## description
+
+check if the cf command is present, if its not, install it and all of its dependencies.
+
+## Args
+
+no args
+
+# copy-login-gov-keypair.sh
+
+## usage
+
+```bash
+./scripts/copy-login-gov-keypair.sh
+```
+
+## Description
+
+Copies Login.gov JWT_KEY + JWT_CERT from one Cloud.gov application to another.
+
+## Args
+
+`SOURCE_APP` The app to copy keys from.
+`DEST_APP` The app to copy keys to.
+
+# deploy-backend.sh
+
+## description
+
+The deployment strategy you wish to employ ( rolling update or setting up a new environment)
+DEPLOY_STRATEGY=${1}
+
+The application name  defined via the manifest yml for the frontend
+CGHOSTNAME_BACKEND=${2}
+
+
+# deploy-frontend.sh
+
+## Args
+
+```bash
+
+# The deployment strategy you wish to employ ( rolling update or setting up a new environment)
+DEPLOY_STRATEGY=${1}
+
+# The application name  defined via the manifest yml for the frontend
+CGHOSTNAME_FRONTEND=${2}
+```
+
+# deploy-infrastructure-dev.sh
+
+Requires installation of jq - https://stedolan.github.io/jq/download/
+
+## Args
+
+no args
+
+# deploy-infrastructure-staging.sh
+
+Requires installation of jq - https://stedolan.github.io/jq/download/
+
+## Args
+
+no args
+
+# docker-check.sh
+
+
+## Usage
+
+```bash
+./scripts/docker-check.sh
+```
+## Args
+
+no args
+
+## Description
+
+Used during ci/cd to check if docker is installed in the environment. 
+It installs it if it is not.
+
+# docker-compose-check.sh
+
+Used during ci/cd to check if docker compose is installed in the environment. 
+It installs it if it is not.
+
+## Args
+
+no args
+
+# git-secrets-check.sh
+
+ensure we have correct configs in place
+grep will return non-zero code if nothing found, failing the build
+
+## Args
+
+no args
+
+# localstack-setup.sh
+
```

Comment:
Missing usage? I thought this had arguments or are they supplied via .env?

Reply:
They are from env vars

Reply:
do we want to cover ENV vars in the readme seperately @abottoms-coder ?

Reply:
> do we want to cover ENV vars in the readme seperately @abottoms-coder ?


I want the arguments for a given script to be explained in the section for that script like you did in `set-backend-env-vars.sh` section. If they're from the environment, please specify that. Please let me know if you have any questions.